src/nl/hannahsten/texifyidea/editor/StyledTextPasteProvider.kt

```diff
@@ -0,0 +1,212 @@
+import nl.hannahsten.texifyidea.util.Log
+import nl.hannahsten.texifyidea.util.currentTextEditor
+import nl.hannahsten.texifyidea.util.files.isLatexFile
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.TextNode
+import java.awt.datatransfer.DataFlavor
+
+/**
+ * Pastes html and applies text(bf|it)
+ *
+ * @author jojo2357
+ */
+class StyledTextPasteProvider : PasteProvider {
+
+    val openingTags = hashMapOf(
+        "i" to "\\textit{",
+        "b" to "\\textbf{",
+        "u" to "\\underline{",
+        "p" to "",
+        "ol" to "\\begin{enumerate}\n",
+        "ul" to "\\begin{itemize}\n",
+        "li" to "\\item ",
+        "br" to "\n",
+        "sup" to "\\textsuperscript{",
+        "sub" to "\\textsubscript{",
+        "h1" to "\\chapter*{",
+        "h2" to "\\section*{",
+        "h3" to "\\subsection*{",
+        "h4" to "\\subsubsection*{",
+        "h5" to "\\subsubsubsection*{",
+    )
+
+    val closingTags = hashMapOf(
+        "i" to "}",
+        "b" to "}",
+        "u" to "}",
+        "p" to "\n\\par\n",
+        "ol" to "\\end{enumerate}\n",
+        "ul" to "\\end{itemize}\n",
+        "li" to "\n",
+        "a" to "}",
+        "br" to "",
+        "sup" to "}",
+        "sub" to "}",
+        "h1" to "}\n",
+        "h2" to "}\n",
+        "h3" to "}\n",
+        "h4" to "}\n",
+        "h5" to "}\n",
+    )
+
+    val specialOpeningTags = hashMapOf<String, (Element) -> String>(
+        "a" to { element ->
+            if (element.hasAttr("href"))
+                if (element.attr("href").startsWith("#"))
+                    "\\hyperlink{" + element.attr("href").replace(Regex("^#"), "") + "}{"
+                else
+                    "\\href{" + escapeText(element.attr("href")) + "}{"
+            else if (element.hasAttr("name"))
+                "\\hypertarget{" + element.attr("name") + "}{"
+            else
+                ""
+        }
+    )
+
+    val specialClosingTags = hashMapOf<String, (Element) -> String>(
+        "a" to { element ->
+            if (element.hasAttr("href"))
+                "}"
+            else if (element.hasAttr("name"))
+                "}"
+            else
+                ""
+        }
+    )
+
+    val escapeChars = hashMapOf(
+        "%" to "\\%",
+        "&" to "\\&",
+        "_" to "\\_",
+        "#" to "\\#",
+        "$" to "\\$",
+        "{" to "\\{",
+        "}" to "\\}",
+        "^" to "\\^",
+        "~" to "\\~",
+        "\\" to "\\textbackslash",
+        "−" to "-"
+    )
+
+    override fun isPastePossible(dataContext: DataContext): Boolean {
+        val file = dataContext.getData(PlatformDataKeys.PSI_FILE) ?: return false
+        if (file.isLatexFile().not()) return false
+        if (ShiftTracker.isShiftPressed()) return false
+
+        val pasteData = dataContext.transferableHtml() ?: return false
+        Log.warn("Attempting to paste $pasteData")
```

Comment:
Don't forget to remove all the warn logs (or better, change them to info) to avoid spam

Reply:
That is for debug at the moment since I don't know how to turn on info (or at least I wasnt getting prints with info)

Reply:
See the documentation for `Log.kt`.
My point is that users use idea.log for debugging other things as well, so we don't want to flood it with TeXiFy messages

Reply:
Still an issue